### PR TITLE
DSP-924 Never Import from index.ts Internally

### PR DESCRIPTION
**Pull request by mpro7**
resolves DSP-924

### CODE REVIEW

File: projects/dsp-ui/src/lib/search/advanced-search/advanced-search.component.ts [hunk 1]
```diff
@@ -21,9 +21,9 @@ import {
     ResourcePropertyDefinition
 } from '@dasch-swiss/dsp-js';
 import { Subscription } from 'rxjs';
-import { NotificationService } from '../../action';
-import { DspApiConnectionToken } from '../../core';
-import { SearchParams } from '../../viewer';
+import { NotificationService } from '../../action/services/notification.service';
+import { DspApiConnectionToken } from '../../core/core.module';
+import { SearchParams } from '../../viewer/';
```
Comment (tobiasschweizer) [inline projects/dsp-ui/src/lib/search/advanced-search/advanced-search.component.ts:26]: I think this should be `
import { SearchParams } from '../../viewer/views/list-view/list-view.component';`

**Review by tobiasschweizer — COMMENTED** [review]
I think a thing that is difficult is the fact that we have one npm package (lib) consisting of several Angular modules.

> To make library code reusable you must define a public API for it. This "user layer" defines what is available to consumers of your library. A user of your library should be able to access public functionality (such as NgModules, service providers and general utility functions) through a single import path.

> The public API for your library is maintained in the public-api.ts file in your library folder. Anything exported from this file is made public when your library is imported into an application. Use an NgModule to expose services and components

https://angular.io/guide/creating-libraries#getting-started

And since we have only on npm package, I think that also inside the lib we should import from the file (and not from the `index.ts` representing the module).

**Comment by tobiasschweizer** [comment]
Maybe we could think about making a sub-entry for each module, https://medium.com/@tomastrajan/the-best-way-to-architect-your-angular-libraries-87959301d3d3. However, I would first have to understand this in general.

File: projects/dsp-ui/src/lib/search/advanced-search/advanced-search.component.ts (hunk 1 above)
Comment (mpro7) [reply projects/dsp-ui/src/lib/search/advanced-search/advanced-search.component.ts]: Thanks, somehow missed it. Done in c7971bd.

**Comment by mpro7** [comment]
> Maybe we could think about making a sub-entry for each module, https://medium.com/@tomastrajan/the-best-way-to-architect-your-angular-libraries-87959301d3d3. However, I would first have to understand this in general.

As you wrote, and as we spoke. This could be a second step.

**Comment by tobiasschweizer** [comment]
> As you wrote, and as we spoke. This could be a second step.

Indeed :-)

I am reading this now: https://medium.com/angular-in-depth/improve-spa-performance-by-splitting-your-angular-libraries-in-multiple-chunks-8c68103692d0

**Review by tobiasschweizer — APPROVED** [review]
Ok, looks good!

Let's learn about splitting up the lib in different sub-entries per module.

**Pull request merged**